### ISSUE TITLE
[rancher-compose] docker-compose volumes: section causes panic

### ISSUE THREAD

**Issue by vthyng** [rancher/rancher#7771]
**Rancher Versions:**
rancher-compose: 0.12.2 and 0.12.2-rc1
Server: 1.3.1 and 1.4
healthcheck:
ipsec:
network-services:
scheduler:
kubernetes (if applicable):

**Docker Version:**
1.12.3

**OS and where are the hosts located? (cloud, bare metal, etc):**
VM in vCenter, but Not Applicable at this point for this issue as far as I can tell.

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle but Not Applicable at this point of the process

**Steps to Reproduce:**
Note docker's documentation that I was referencing.  See the v2 example under external (this is not a v3 format feature)
https://docs.docker.com/compose/compose-file/#volume-configuration-reference
Add a volumes: section to docker-compose.yml file such as
volumes:
   postgres_data:

**Results:** 
rancher-compose \
--url https://rancher.west.isilon.com/v2-beta \
--access-key C01D9F4E1C1C72AC3DBE \
--secret-key iibJLpwbAQgm6WKbKijxrxcnddE7SNkkUFGD2ogG \
up -d --pull --force-upgrade --confirm-upgrade 
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x53a1e3]

goroutine 1 [running]:
panic(0x9548a0, 0xc420014130)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/rancher/rancher-compose/vendor/github.com/urfave/cli.HandleAction.func1(0xc4204cc9d0)
	/go/src/github.com/rancher/rancher-compose/vendor/github.com/urfave/cli/app.go:472 +0x29e
panic(0x9548a0, 0xc420014130)
	/usr/local/go/src/runtime/panic.go:458 +0x243
github.com/rancher/rancher-compose/rancher.(*RancherVolumesFactory).Create(0xc42003a468, 0xc4202d953f, 0x7, 0xc4202d94a0, 0xc4202ebf60, 0x1, 0x12, 0xc420322500, 0x28b, 0x48b)
	/go/src/github.com/rancher/rancher-compose/rancher/volume.go:21 +0x1b3
github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/project.(*Project).load(0xc4200a8700, 0xa16a5a, 0x12, 0xc420322500, 0x28b, 0x48b, 0xc4201ac2d0, 0x4104b8)
	/go/src/github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/project/project.go:247 +0x4ae
github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/project.(*Project).Parse(0xc4200a8700, 0x0, 0x0)
	/go/src/github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/project/project.go:112 +0x15a
github.com/rancher/rancher-compose/rancher.NewProject(0xc4200a68c0, 0xa0f66b, 0x4, 0x301)
	/go/src/github.com/rancher/rancher-compose/rancher/project.go:46 +0x1ad
github.com/rancher/rancher-compose/app.(*ProjectFactory).Create(0xd13b88, 0xc4202db7c0, 0x0, 0x8, 0x8, 0xc4202ebde0)
	/go/src/github.com/rancher/rancher-compose/app/app.go:66 +0x7a0
github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/cli/app.WithProject.func1(0xc4202db7c0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/cli/app/app.go:45 +0x4e
reflect.Value.call(0x918ea0, 0xc4202ead60, 0x13, 0xa0f2f3, 0x4, 0xc4204cc970, 0x1, 0x1, 0x485128, 0x9f7fa0, ...)
	/usr/local/go/src/reflect/value.go:434 +0x5c8
reflect.Value.Call(0x918ea0, 0xc4202ead60, 0x13, 0xc4204cc970, 0x1, 0x1, 0x315fab6c, 0x140, 0x140)
	/usr/local/go/src/reflect/value.go:302 +0xa4
github.com/rancher/rancher-compose/vendor/github.com/urfave/cli.HandleAction(0x918ea0, 0xc4202ead60, 0xc4202db7c0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher-compose/vendor/github.com/urfave/cli/app.go:481 +0x1e0
github.com/rancher/rancher-compose/vendor/github.com/urfave/cli.Command.Run(0xa0ec07, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0xa185f6, 0x15, 0x0, ...)
	/go/src/github.com/rancher/rancher-compose/vendor/github.com/urfave/cli/command.go:186 +0xc26
github.com/rancher/rancher-compose/vendor/github.com/urfave/cli.(*App).Run(0xc4202bc180, 0xc42000a0c0, 0xc, 0xc, 0x0, 0x0)
	/go/src/github.com/rancher/rancher-compose/vendor/github.com/urfave/cli/app.go:235 +0x60c
main.cliMain()
	/go/src/github.com/rancher/rancher-compose/main.go:89 +0xf86
main.main()
	/go/src/github.com/rancher/rancher-compose/main.go:27 +0x5e

**Comment by vthyng** [comment]
I was able to resolve this error by adding external: false
```
volumes:
   postgres_data:
      external: false
```

**Comment by gaieges** [comment]
I can confirm the same problem on rancher-compose v0.12.2, however I can also reproduce it if there is ANY volume entries (eg.):

```yaml
volumes:
  nginxLetsEncrypt:
```

**Comment by joshwget** [comment]
Available in Rancher Compose Executor v0.13.0, Rancher Compose v0.12.3-rc2, and Rancher CLI v0.5.0-rc1.

**Comment by sangeethah** [comment]
Tested with rancher-server version - v1.5.0-rc3

Able to use the following yml to create services in the following ways:
1. Creating stack from UI providing this yml
2. Creating stack from rancher cli - rancher-v0.5.0-rc1
3. Creating stack from rancher compose - rancher-compose-v0.12.3-rc2

```
version: '2'
volumes:
  test1:
services:
  s1:
    image: ubuntu:14.04.3
    stdin_open: true
    volumes:
    - test1:/myvol
    tty: true
    labels:
      io.rancher.container.pull_image: always
```
